### PR TITLE
Batch upload

### PR DESCRIPTION
**Pull request by officialpranav**
Note: this version uses the pre-golive version of maximo. Will need to be updated once golive happens.

### CODE REVIEW

File: assets/exceljs.js [hunk 1]
```diff
@@ -300,6 +302,27 @@ class Spreadsheet {
         });
         postMessage(['result', [meters.slice(1), observations.slice(1)]]);
     }
+
+    async getCellLoc(value) {
+        // returns all the english descriptions on a workbook
```
Comment (jonathanmajh) [inline assets/exceljs.js:307]: dead code?

File: renderer/item_main.js [hunk 2]
```diff
@@ -234,6 +289,172 @@ function openExcel() {
...
+            colLoc.maximo=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='SERIES'){
+            colLoc.series=i;
+        }
+        //console.log(validParams)
+    }
+    if(validParams<5){
+        document.getElementById("batch-upload-status-text").innerHTML=`Table is missing ${5-validParams} column(s)! Table will not be uploaded.`;
+        return;
```
Comment (jonathanmajh) [inline renderer/item_main.js:334]: needs to return names of expected column headers

**Review by jonathanmajh — COMMENTED** [review]
maybe a copy button that contains the headers?

**Comment by jonathanmajh** [comment]
Replace the share button under item creation validation, remove old code related to upload
image

File: renderer/item_main.js (hunk 2 above)
Comment (jonathanmajh) [reply renderer/item_main.js:334]: maybe also add a button to copy what we want the headers to be

File: renderer/item_main.js [hunk 3]
```diff
@@ -234,6 +289,172 @@ function openExcel() {
...
+        } else if(cell.innerHTML.toUpperCase()==='COMMODITY GROUP' || cell.innerHTML.toUpperCase()==='COMM GROUP'){
+            colLoc.commGroup=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='GL CLASS'){
+            colLoc.glClass=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='MAXIMO' || cell.innerHTML.toUpperCase()==='ITEM NUMBER'){
+            colLoc.maximo=i;
+            validParams++;
+        } else if(cell.innerHTML.toUpperCase()==='SERIES'){
```
Comment (jonathanmajh) [inline renderer/item_main.js:327]: would it be easier to just say the columns must be in a specific order?

Comment (officialpranav) [reply renderer/item_main.js:327]: ya we can say it in the docs, it was just easier to code it this way

**Pull request merged**